### PR TITLE
Arnold standard surface PBR parsing

### PR DESCRIPTION
**Pull request by tntmeijs**
I've added a parser for Arnold standard surface PBR shaders.
Known issues:
- Only parses shader upon initialization.
- Runs out of memory fairly quickly.
- Fails to parse sub color nodes (needs more investigation).

## Screenshots (if appropriate):
image
image

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/plugin/parsers/material_parser.cpp [hunk 1]
```diff
@@ -1,9 +1,506 @@
...
+				}
+				break;
+
+				// All faces use the same material
+			case 1:
+				{
+					auto surface_shader = GetSurfaceShader(shaders[0]);
+
+					// Invalid surface shader
+					if (!surface_shader.has_value())
```
Comment (sainsay) [inline src/plugin/parsers/material_parser.cpp]: multiple if statements do not conform style guide. please use { }

File: src/plugin/parsers/material_parser.cpp [hunk 2]
```diff
@@ -1,9 +1,506 @@
...
+		// References to the shaders used on the meshes
+		MObjectArray shaders;
+
+		// Indices to the materials in the object array
+		MIntArray material_indices;
+
+		// Get all attached shaders for this instance
+		mesh.getConnectedShaders(instance_index, shaders, material_indices);
+
+		switch (shaders.length())
```
Comment (sainsay) [inline src/plugin/parsers/material_parser.cpp:99]: consider putting code in a function for big code blocks in switch statement if possible.

File: src/plugin/parsers/material_parser.cpp [hunk 3]
```diff
@@ -1,9 +1,506 @@
...
+const MPlug wmr::MaterialParser::GetPlugByName(const MObject& node, MString name)
+{
+	return MFnDependencyNode(node).findPlug(name);
+}
+
+const std::optional<MPlug> wmr::MaterialParser::GetSurfaceShader(const MObject& node)
+{
+	MPlug shader_plug = MFnDependencyNode(node).findPlug(MayaMaterialProps::surface_shader, true);
+
+	if (!shader_plug.isNull())
```
Comment (sainsay) [inline src/plugin/parsers/material_parser.cpp:289]: not conform code style guide. { }

File: src/plugin/parsers/material_parser.hpp [hunk 4]
```diff
@@ -1,14 +1,117 @@
...
 
 	private:
+		// std::pair
+		//    first: MObject, connected lambert plug
+		//    second: MObject, MFnMesh.object()
+		std::vector<std::pair<MObject, MObject>> mesh_material_relations;
 
+		Renderer& m_renderer;
 	};
 }
```
Comment (sainsay) [inline src/plugin/parsers/material_parser.hpp]: no newline at end of file

**Review by sainsay — CHANGES_REQUESTED** [review]

File: src/plugin/parsers/material_parser.hpp (hunk 4 above)
Comment (tntmeijs) [reply src/plugin/parsers/material_parser.hpp]: Done

File: src/plugin/parsers/material_parser.cpp (hunk 1 above)
Comment (tntmeijs) [reply src/plugin/parsers/material_parser.cpp]: All fixed!

File: src/plugin/parsers/material_parser.cpp (hunk 2 above)
Comment (tntmeijs) [reply src/plugin/parsers/material_parser.cpp:99]: Okay, I'll make sure to do it.

File: src/plugin/parsers/material_parser.cpp (hunk 3 above)
Comment (tntmeijs) [reply src/plugin/parsers/material_parser.cpp:289]: Done.

**Review by sainsay — APPROVED** [review]
changes requested have been implemented

**Pull request merged**